x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/physical/FieldExtractExec.java

```diff
@@ -7,39 +7,71 @@
 
 package org.elasticsearch.xpack.esql.plan.physical;
 
+import org.elasticsearch.TransportVersions;
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.xpack.esql.core.expression.Attribute;
 import org.elasticsearch.xpack.esql.core.expression.AttributeSet;
 import org.elasticsearch.xpack.esql.core.tree.NodeInfo;
 import org.elasticsearch.xpack.esql.core.tree.NodeUtils;
 import org.elasticsearch.xpack.esql.core.tree.Source;
+import org.elasticsearch.xpack.esql.io.stream.PlanStreamInput;
+import org.elasticsearch.xpack.esql.io.stream.PlanStreamOutput;
 
+import java.io.IOException;
 import java.util.ArrayList;
-import java.util.Collection;
 import java.util.List;
 import java.util.Objects;
 import java.util.Set;
 
-import static java.util.Collections.emptySet;
-
 public class FieldExtractExec extends UnaryExec implements EstimatesRowSize {
+    public static final NamedWriteableRegistry.Entry ENTRY = new NamedWriteableRegistry.Entry(
+        PhysicalPlan.class,
+        "FieldExtractExec",
+        FieldExtractExec::new
+    );
+
     private final List<Attribute> attributesToExtract;
     private final Attribute sourceAttribute;
     // attributes to extract as doc values
     private final Set<Attribute> docValuesAttributes;
 
     private List<Attribute> lazyOutput;
 
-    public FieldExtractExec(Source source, PhysicalPlan child, List<Attribute> attributesToExtract) {
-        this(source, child, attributesToExtract, emptySet());
-    }
-
     public FieldExtractExec(Source source, PhysicalPlan child, List<Attribute> attributesToExtract, Set<Attribute> docValuesAttributes) {
         super(source, child);
         this.attributesToExtract = attributesToExtract;
         this.sourceAttribute = extractSourceAttributesFrom(child);
         this.docValuesAttributes = docValuesAttributes;
     }
 
+    private FieldExtractExec(StreamInput in) throws IOException {
+        this(
+            Source.readFrom((PlanStreamInput) in),
+            ((PlanStreamInput) in).readPhysicalPlanNode(),
+            in.readNamedWriteableCollectionAsList(Attribute.class),
+            in.getTransportVersion().onOrAfter(TransportVersions.ESQL_FIELD_EXTRACT_DOC_VALUES)
+                ? in.readCollectionAsSet(i -> i.readNamedWriteable(Attribute.class))
+                : Set.of()
```

Comment:
This was missing before I got here. I'm not sure what, if anything, is broken without it. But if we're going to serialize, let's do it right.

Reply:
This set is only planned and used in local physical planning on a data node, and so we decided not to bother serializing it. It will always be an empty set on the coordinator node, and if we add it to the serialization, we are simply serializing an empty set all the time.

The use case for this feature was spatial point extraction from doc-values instead of source. This decision is made at local physical planning time based on the existence of an appropriate spatial aggregation in the data nodes plan, one that erases the need for high precision original source point field extraction. It is conceivable in the longer term that we come up with scenarios where we decide to plan this at logical planning time, and would then need to serialize this field (and presumably make a few other serialization changes too, because it is not only field extraction that is flagged with this knowledge).

Reply:
My vote would be to only change the serialization when we actually need the change, but would not block this change if we feel it is decent defence against a possible future need. But the knowledge of doc values is also saved in SpatialRelatesFunction (when there is a spatial predicate on the data node before the aggregation) and the SpatialAggregateFunction (the one that enables doc-values in the first place). Perhaps we want to add the docs values to the serialization there too? My vote would be to do all three places in the same PR and that it not be this PR.

Reply:
> This set is only planned and used in local physical planning on a data node, and so we decided not to bother serializing it. It will always be an empty set on the coordinator node, and if we add it to the serialization, we are simply serializing an empty set all the time.

In that case I'll remove the serialization and document this. It's very surprising when serializable stuff doesn't serialize *all* of itself. especially if that stuff is used in `equals` and `hashCode`. But it's allowed. It just need a big sign.